### DISCUSSION TITLE
Help Patching 023000ee

### DISCUSSION THREAD

**Discussion by DipeshChunilal** [Mr-MIBonk/M.I.B._More-Incredible-Bash#226] [Q&A] [answered]
Hi all
Recently i updated my firmware on my A4 B9 2017 from MHI2Q_ER_AUG22_P4212 to P5092 MU1329 sucessfully, right after managed to update POI and SpeedCams as well but unfortunately get stuck with nav as i believe i dont have the EU lifetime activation keys.
I have managed to download MIB instructions and installed but not patching the activation keys.
i have gone through all the discussion here in regards that and tried what everyone done but still no luck.
wondering if anyone can help?
i am attaching some photos and one last thing nothing on the backup log of MIB
addFecs on patch file for my firmware is fine as the 023000ee is there

![IMG_2793](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/6f411d66-baf1-4c65-a1ad-7e18a0aedff8)
![IMG_2794](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/3120a969-829f-4a70-b3a7-1606f8161cb5)
![IMG_2795](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/c18b0b04-f433-46f3-a85e-88898951fe20)
![IMG_2796](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/a48e0c71-0236-406f-b499-c6d1fb7092bc)
![IMG_2797](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/c8eb5d4f-28d1-427f-bae5-0072b52383cc)
![IMG_2798](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/4246da60-c22e-4d3d-a24d-050dcc8e0b7c)
![IMG_2799](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/134564640/047f93c2-0219-42b1-912a-efcf8419dc04)

**Comment by archercj** [comment]
Looks like I've got a very similar car to you (mines and Audi S4 B9) and the same firmware.

Check what is happening in your log files on you SD Card - backup\logs directory.

When I first run the FEC patch SUCCESSFULLY on an earlier firmware I got this:

```
_[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...

There will be no further output until flash is finished!


Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f0fc84 (15793284)
End addr:   0x01aafc84
*** erasing:     0x00BA0000 .. 0x01AAFC83 [01ABFFFF] ......................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AAFC83            ......................................................................................................................... 

OK: Flash checked and valid
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0d53 ( 3738964)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9bf07 ( 6668040)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1aafc83 (15793284)|93...214 (121):LOCKED
4 image type=4(DSP     ) 3d00000...3dc61b3 (  811444)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1aafc83 (15793284)|93...214 (121):LOCKED
0x00ba0000 - 84FCF000

dump RCC to logs - DURATION: ~3 minutes......... -> done :-)

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2Q_ER_AUG22_P3638_MU0917_PATCH... -> done :-)
copy EL file from /patches/MHI2Q_ER_AUG22_P3638_MU0917_PATCH... -> done :-)

-----------
timestamp: 1970_01_01_00_06_45

MMX-showimage---->
Reboot Unit..._


BUT I then proceeded to update the MMI f/w (successfully may I add) but when I the reran the MBI FEC patch I got a half bricked unit and a log that ended:

**[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...

There will be no further output until flash is finished!**
```
Had to seek help after managing to enter the MMI system via a laptop connection but couldnt get the MIB FEC patch to work again. 

Funny there are issues with the same Audi firmware (when a previous version of the firmware worked fine)

**Reply by DipeshChunilal** [reply]
theres no files on SD Card - backup\logs directory.
its completely empty
what should i do? have i bricked my mmi or theres a solution for it, shall i update the firmware again as theres P5152 available then try mib again?
thank you

**Reply by harman-f** [reply]
no files, SD is write protected

**Reply by DipeshChunilal** [reply]
i did access all the files on SD but even backup logs but was nothing recorded in it

**Comment by archercj** [comment, marked as answer]
If the MMI is operational it isnt bricked.

I suggest you download the latest MIB from here - theres 3.6.0 available now

Looking at your screen shots I reckon youve done this ok but key tips are:

Make sure the SD card is formatted to FAT32
change the _Swdlautorun.txt file to Swdlautorun.txt (that means it will automatically run)
Only use SD1 

Oh and make sure you have the correct backups folder (its a patched version of the stock file) in the backup folder. You will need MU1329-MHI2Q_ER_AUG22_P5092-JTB-07415.11.1611900687

It can be found here - https://mibsolution.one (you might need to register). Its xipped full of other patched firmware. Just copy that folder to your backup folder of your sdcard.

Like I say proceed with caution. The cars original stock MU0917-MHI2Q_ER_AUG22_P3638-JTB-07415.11.1611900687 worked fine with the FEC patch but after I updated the MMI (and needed to reapply the FEC patch) when I tried it using MU1329-MHI2Q_ER_AUG22_P5092-JTB-07415.11.1611900687 I ran into difficulties with the patch process.

BUT I believe it may have been my inpatience. When the screen goes blank just leave it for ages (like 30 minutes).

**Reply by DipeshChunilal** [reply]
I appreciate the help.
unfortunately i can't find MIB 3.6.0 and once found do i install it again or just run it?
also unable to download any files from backup that contains MU1329-MHI2Q_ER_AUG22_P5092-JTB-07415.11.1611900687 from https://mibsolution.one/ and no option to register as i logged in as guest and pass guest as well

**Comment by archercj** [comment]
If you go to the Code section (to the left of the Discussions tab you are on now) you will see a Green '<> Code' button. Select that and you will see a download option. That is downloading the laest code.

But at the moment ignore that and use the version you alread have on your SDCard.

And I'm being a plonker. You dont need to register. Guest and guest is correct. If you look at the folders to the left of the page select the M.I.B folder and in there there is  file called 20230412_M.I.B_Patches. Right click the blue icon to the left of the filename and it will allow you to download.

Then unzip it. Its got loads of folders with patches. You just need the folder MHI2Q_ER_AUG22_P5092_MU1329_PATCH and place the whole folder in the patches folder on your SDCard

**Reply by DipeshChunilal** [reply]
will give a try now. fingers crossed

**Reply by DipeshChunilal** [reply]
tha'ts what i did initially

**Comment by DipeshChunilal** [comment]
Second attempt no mib?

**Comment by DipeshChunilal** [comment]
i have now sorted. what i did was download the latest mib 3.6.0 and add a patch for my car and install it then done the back up and right after patch up and it worked. checked to the nav key was there and now have updated nav with 2023/24 full europe as no englad alone file.

thank you @archercj and @harman-f

**Reply by archercj** [reply]
Brill. Thankfully it worked for you!

**Comment by BooBoo124** [comment]
Sorry to hijack the thread but I can't get an answer on a question I've raised. 
I have patched MHI2Q-AUG22-P4212 successfully with MIB a while back. If I up date straight to the latest firmware MHI2Q_ER_AUG22_P5092 will it remove 23000ee fec, and I need to patch again or can I update and patch in one go, or do I need to update and then patch again?

Cheers
